app/scripts/lib/AbstractPetnamesBridge.ts

```diff
@@ -28,13 +32,14 @@ enum SyncDirection {
 // A list of changes, grouped by type.
 type ChangeList = Record<ChangeType, PetnameEntry[]>;
 
-type AllowedEvents = NameStateChange;
+type AllowedEvents = NameStateChange | AccountsControllerChangeEvent;
+type AllowedActions = AccountsControllerListAccountsAction;
```

Comment:
Let's try not to mix account events and actions and actions in this file. It makes our abstract class too aware of its subclasses' details. How about we fix this by:

1. Letting AbstractPetnamesBridge work with a more generic messenger.
2. Allowing PetnamesBridgeMessenger to use template parameters for actions.
3. Using a different messenger for the subclass.

Or, is there another approach we could consider?

Reply:
Sure, we can make it work with a more generic messenger